lego/apps/users/models.py

```diff
@@ -495,26 +519,69 @@ class Penalty(BasisModel):
     source_event = models.ForeignKey(
         "events.Event", related_name="penalties", on_delete=models.CASCADE
     )
+    activation_time = models.DateTimeField("date created", null=True, default=None)
 
     objects = UserPenaltyManager()  # type: ignore
 
-    def expires(self):
-        dt = Penalty.penalty_offset(self.created_at) - (
-            timezone.now() - self.created_at
+    def save(self, *args, **kwargs):
+        last_penalty_to_be_expired = Penalty.objects.filter(
+            user=self.user, activation_time__lte=timezone.now()
+        ).order_by("-activation_time")
+
+        self.activation_time = (
+            last_penalty_to_be_expired[0].exact_expiration
+            if len(last_penalty_to_be_expired) != 0
+            else self.created_at
         )
-        return dt.days
+        super().save(*args, **kwargs)
+
+    def default_save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+
+    def expires(self):
+        expirationDate = Penalty.penalty_offset(
+            self.activation_time, weight=self.weight
+        ) - (timezone.now() - self.activation_time)
+        return expirationDate.days
```

Comment:
Took me a while to understand this logic, so I feel like this could be written a bit better. How about something like
expiration_date = activation_time + penalty_offset
days_until_expiry = expiration_date - now
instead? As it is now it's kind of unclear where you are calculating a date and where you are calculating a time interval